app/jobs/worker/cdr_export_job.rb

```diff
@@ -0,0 +1,45 @@
+module Worker
+  class CdrExportJob < ActiveJob::Base
+    queue_as 'cdr_export'
+
+    attr_reader :cdr_export
+
+    def perform(cdr_export_id)
+      @cdr_export = find_cdr_export(id)
+
+      Cdr::Cdr.connection.execute("COPY (#{self.raw_sql}) TO '#{file_path}' WITH CSV HEADER;")
+      # update cdr_export status
+      cdr_export.update!(status: 'Completed')
+
+      # ping callback_url
+      if cdr_export.callback_url
+        params={} #todo: what needed params should be requested? filename, x-accel-redirect?
+        PingCallbackUrlJob.perform_later(cdr_export.callback_url, params)
+      end
+    rescue => _e
+      #todo: when failed?
+      cdr_export.update!(status: 'Failed')
+    end
+
+    private
+
+    #todo: check on ransack availability
+    def raw_sql
+      Cdr::Cdr.select(cdr_export.fields.join(', ')).order('time_start desc').ransack(cdr_export.filters).result.to_sql
```

Comment:
it should be responsibility of CdrExport class, it should be possible to use  different cdr_export object that can have different raw_sql data